build-tools/build-indexes

```diff
@@ -1017,7 +1016,216 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
+					let usedTier = tier;
+					if (usedTier === "(UU)") usedTier = "RU";
+					if (usedTier === "(NU)") usedTier = "PU";
+					if (usedTier === "(PU)") usedTier = "ZU";
+					if (usedTier === "(DUU)") usedTier = "DNU";
+					formatSlices[usedTier] = tiers.length;
+				}
+				if (!tierTable[tier]) continue;
+				tiers.push(['header', tier]);
+				tiers.push(...tierTable[tier]);
+			}
+			//pokemon stats
+			const overrideStats = {};
+			BattleTeambuilderTable[modid].overrideStats = overrideStats;
+			const overrideType = {};
+			BattleTeambuilderTable[modid].overrideType = overrideType;
+			const overrideAbility = {};
+			BattleTeambuilderTable[modid].overrideAbility = overrideAbility;
+			const overrideSecondAbility = {};
+			BattleTeambuilderTable[modid].overrideSecondAbility = overrideSecondAbility;
+			const overrideHiddenAbility = {};
+			BattleTeambuilderTable[modid].overrideHiddenAbility = overrideHiddenAbility;
+			const removeSecondAbility = {};
+			BattleTeambuilderTable[modid].removeSecondAbility = removeSecondAbility;
+			const requiredItem = {};
+			BattleTeambuilderTable[modid].requiredItem = requiredItem;
+			const fullFakemonData = {};
+			BattleTeambuilderTable[modid].fullFakemonData = fullFakemonData;
+			for (const id in modData.Pokedex) {
+				const modEntry = modData.Pokedex[id];
+				const baseEntry = Dex.data.Pokedex[id];
+				const baseType = () => {
+					if (typeof baseEntry !== 'undefined') return baseEntry.types.join('/');
+				};
+				const fakeMon = (typeof baseEntry === 'undefined');
+				if (fakeMon) {
+					fullFakemonData[id] = modEntry;
+				} else {
+					for (const stat in modEntry.baseStats) {
+						if (modEntry.baseStats[stat] !== baseEntry.baseStats[stat]) {
+							if (!overrideStats[id]) overrideStats[id] = {};
+							overrideStats[id][stat] = modEntry.baseStats[stat];
+						}
+					}
+					if (modEntry.types.join('/') !== baseType) {
+						overrideType[id] = modEntry.types.join('/');
+					}
+					if (modEntry.abilities['0'] !== baseEntry.abilities['0']) {
+						overrideAbility[id] = modEntry.abilities['0'];
+					}
+					if (modEntry.abilities['1'] !== baseEntry.abilities['1']) {
+						overrideSecondAbility[id] = modEntry.abilities['1'];
+					}
+					if (modEntry.abilities['H'] !== baseEntry.abilities['H']) {
+						overrideHiddenAbility[id] = modEntry.abilities['H'];
+					}
+					if (modEntry.requiredItem && !baseEntry.requiredItem) {
+						requiredItem[id] = modEntry.requiredItem;
+					}
+				}
+			}
+			//learnsets
+			const learnsets = {};
+			BattleTeambuilderTable[modid].learnsets = learnsets;
+			for (const id in modDex.data.Learnsets) {
+				const learnset = modDex.data.Learnsets[id].learnset;
+				if (!learnset) continue;
+				learnsets[id] = {};
+				for (const moveid in learnset) {
+					learnsets[id][moveid] = '012345678qgp';
```

Comment:
What is this about?

Reply:
A catchall to make sure moves would legal... possibly something I tried when trying to fix an issue and never changed back. If the proper learnsets string is available we could pass that. Move compatibility isn't normally a thing in mods but this would mess that up if it ever was.

Reply:
Just to clarify, another part of the code checks this string for characters related to different generations or offshoots that a move might be from. 

I mentioned a potential problem with move compatibility but that is not an issue; the teambuilder already does not take it into account (the validator handles that). 

The fact that it is loading the list of moves from the mod in question also means that we can assume any moves being added to learnsets[id] should be showing up in the teambuilder no matter what. I verified that it works for past gen mods, gen8 OU-based mods, and natdex-based mods. All of them get the right list of moves. Nothing special had to be done, so I'm pretty sure this will load movesets properly for all mods, provided the mod itself functions properly.